receiver/mongodbatlasreceiver/internal/mongodb_atlas_client.go

```diff
@@ -0,0 +1,104 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+
+	"github.com/mongodb-forks/digest"
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap"
+)
+
+// This struct wraps the official MongoDB Atlas client to manage pagination
+// and mapping to OpenTelmetry metric and log structures.
+type MongoDBAtlasClient struct {
+	log    *zap.Logger
+	client *mongodbatlas.Client
+}
+
+func NewMongoDBAtlasClient(
+	publicKey string,
+	privateKey string,
+	log *zap.Logger,
+) (*MongoDBAtlasClient, error) {
+	t := digest.NewTransport(publicKey, privateKey)
+	tc, err := t.Client()
+	if err != nil {
+		return nil, err
+	}
+	client := mongodbatlas.NewClient(tc)
+	return &MongoDBAtlasClient{
+		log,
+		client,
+	}, nil
+}
+
+// Check both the returned error and the status of the HTTP response
+func checkMongoDBClientErr(err error, response *mongodbatlas.Response) error {
+	if err != nil {
+		return err
+	}
+	if response != nil {
+		return mongodbatlas.CheckResponse(response.Response)
+	}
+	return nil
+}
+
+func hasNext(links []*mongodbatlas.Link) bool {
+	for _, link := range links {
+		if link.Rel == "next" {
+			return true
+		}
+	}
+	return false
+}
+
+// Return a list of all organizations available with the supplied credentials
+func (s *MongoDBAtlasClient) Organizations(ctx context.Context) []*mongodbatlas.Organization {
+	allOrgs := make([]*mongodbatlas.Organization, 0)
+	page := 1
+
+	for {
+		orgs, hasNext, err := s.getOrganizationsPage(ctx, page)
+		page++
+		if err != nil {
+			// Error was reported in getOrganizationsPage
+			break // Stop, returning what we have (probably empty slice)
```

Comment:
Given that you are swallowing the error here, move the logging statement you have in the `getOrganizationsPage` to this place here. Make sure to use the `debug` level instead. Ideally, this failure would also be recorded as a gauge.

Reply:
I was just trying to toss in a metric here too, but I was having a difficult time finding an example of another receiver that does this. I gave myself a TODO for the moment while I look for good examples to emulate. I'll add a metric on the next PR with the client as I get it figured out. It should probably be a sum error count with the failed call as an attribute?

Reply:
Just keep this as a todo somewhere, as I'm not sure we have sorted out observability for the collector yet (or have we, @bogdandrutu?)